utils/test.go

```diff
@@ -8,14 +8,22 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func HexToFelt(t testing.TB, hex string) *felt.Felt {
+func HexTo[T ~[4]uint64](t testing.TB, hex string) *T {
 	t.Helper()
+
 	f, err := new(felt.Felt).SetString(hex)
 	require.NoError(t, err)
-	return f
+	x := T(*f)
+	return &x
```

Comment:
Should it be `(*T)(f)`?

Reply:
That would be ideal but it seems Go doesn't support that. We can get around with `unsafe.Pointer(...)` but I don't think performance is so critical for this function